src/Entity/OgMembership.php

```diff
@@ -320,6 +320,22 @@ public function preSave(EntityStorageInterface $storage) {
       throw new \LogicException('OG membership can not be created for an empty or anonymous user.');
     }
 
+    if (!$this->get('entity_id')->value) {
+      // Group was not set.
+      throw new \LogicException('Membership cannot be set for an empty or an unsaved group.');
```

Comment:
I'm facing some issues with the API, I was able to see it working for example on a user, but even following the [docs](https://www.drupal.org/node/2015613) causes an error:

```php
    $definition = DataDefinition::create('integer')
      ->setConstraints(['Range', ['min' => 5]]);

    $typed_data = \Drupal::typedDataManager()->create($definition, 10);
    $violations = $typed_data->validate();
```

results with:

```
Drupal\Component\Plugin\Exception\PluginNotFoundException: The "0" plugin does not exist. in Drupal\Core\Plugin\DefaultPluginManager->doGetDefinition() (line 52 of /Library/WebServer/Documents/d8_dev/core/lib/Drupal/Component/Plugin/Discovery/DiscoveryTrait.php).
```

Furthermore, when I tried to create an `OgMembership`  and `$membership->validate()` on the existing 8.x-1.x I got

```
Drupal\Component\Plugin\Exception\PluginNotFoundException: The "entity:og_membership:og_membership_type_default" plugin does not exist. in Drupal\Core\Plugin\DefaultPluginManager->doGetDefinition()
```

Can you maybe spot what I'm doing wrong?